server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/SingleSeriesCompactionExecutor.java

```diff
@@ -101,6 +121,9 @@ public void execute() throws IOException {
       List<ChunkMetadata> chunkMetadataList = readerListPair.right;
       for (ChunkMetadata chunkMetadata : chunkMetadataList) {
         Chunk currentChunk = reader.readMemChunk(chunkMetadata);
+        if (this.chunkWriter == null) {
+          constructChunkWriterFromReadChunk(currentChunk);
+        }
```

Comment:
In the write-whole-chunk path, the chunk writer seems not used (FileWriter is used instead). So maybe it is too early to initialize it here. Maybe you could add a getter of chunkWriter and let the getter do the initialization.